sklearn/ensemble/forest.py

```diff
@@ -378,7 +378,7 @@ def feature_importances_(self):
 # ForestClassifier or ForestRegressor, because joblib complains that it cannot
 # pickle it when placed there.
 
-def accumulate_prediction(predict, X, out, lock):
```

Comment:
add a doctring rather than changing the public API.

Reply:
It looks like it should have been private in the first place because it's only used locally:

```
~/dev/scikit-learn(branch:master) » ag accumulate_prediction 
sklearn/ensemble/forest.py
381:def accumulate_prediction(predict, X, out, lock):
587:            delayed(accumulate_prediction)(e.predict_proba, X, all_proba, lock)
694:            delayed(accumulate_prediction)(e.predict, X, [y_hat], lock)

```


We're having similar discussions in #11416 and #11407.

Reply:
This LGTM overall. @NicolasHug I would suggest adding a doctring here. We should be very careful when making public functions private.

Reply:
This function was only introduced in 0.19, changed its signature in 0.19.1, and is not useful to users. We really don't need to worry too much about making it private.